packages/language-server/src/server.ts

```diff
@@ -164,6 +170,43 @@ documents.onDidChangeContent(async (changeEvent) => {
   }
 });
 
+connection.onCodeAction((params) => {
+  const docUri = params.textDocument.uri;
+  const codeActions: CodeAction[] = [];
+  const textDocument = documents.get(docUri);
+  if (textDocument === undefined) {
+    return undefined;
+  }
+
+  const diagnostics = params.context.diagnostics;
+  forEach(diagnostics, (_) => {
+    const codeAction = getQuickFixCodeAction(textDocument, _);
+    if (codeAction !== undefined) {
+      codeActions.push(codeAction);
+    }
+  });
+
+  return codeActions;
+});
+
+connection.onExecuteCommand(async (params) => {
+  if (params.arguments === undefined) {
+    return;
+  }
+
+  const textDocument = documents.get(params.arguments[0]);
+  if (textDocument === undefined) {
+    return;
+  }
+
+  const textEdit = executeCommand(textDocument, params);
```

Comment:
I think there may be too many assumptions here.
- Should we check more clearely what type of command it before executing it?
- Would all commands we ever implement will return textEdits?